src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,75 @@
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import { getProduct } from '../../helpers/getProduct';
+import { Product } from '../../types/Product';
+import { Loader } from '../Loader/Loader';
+import './Breadcrumbs.scss';
+
+export const Breadcrumbs: React.FC = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [product, setProduct] = useState<Product>();
+  const { pathname } = useLocation();
+
+  const pathNameElement = pathname.slice(1).split('/');
+  const category = `${pathNameElement[0][0]
+    .toLocaleUpperCase()}${pathNameElement[0].slice(1)}`;
+  const productId = pathNameElement[1] || null;
+
+  useEffect(() => {
+    if (productId) {
+      setIsLoading(true);
+      try {
+        (async () => {
```

Comment:
you can use then to work with the result of a promise, in this way you don't need to add this wrapper, and your code will looks more concisely

Reply:
done